test/webpack-integration.test.js

```diff
@@ -42,87 +42,41 @@ describe('Webpack Integration Tests', () => {
   });
 
   it('calls cssProcessor with correct arguments', (done) => {
-    fs.readFile(__dirname + '/default/default.css', (err, data) => {
-      const expectedCss = data.toString();
-      const destination = 'tmp.css';
-      const cssProcessor = {
-        process: (css, options) => {
-          const assembledOptions = Object.assign({ from: destination, to: destination }, options);
-          expect(options).toEqual(assembledOptions);
-          expect(css).toEqual(expectedCss);
-          return Promise.resolve({ css });
-        }
-      };
-      const cssProcessorOptions = { discardComments: { removeAll: true } };
-      const plugin = new OptimizeCssAssetsPlugin({
-        cssProcessor,
-        cssProcessorOptions
-      });
-      const config = Object.assign(defaultConfig, {plugins: [plugin, new ExtractTextPlugin(destination)]});
-      webpack(config, (err, stats) => {
-        if (err) return done(err);
-        if (stats.hasErrors()) return done(new Error(stats.toString()));
-        done();
-      });
+    const destination = 'destination.css';
+    const expectedCss = readFileOrEmpty(__dirname + '/util/default.css');
+    const cssProcessor = {
+      process: (actualCss, options) => {
+        expect(options).toEqual(expect.objectContaining(cssProcessorOptions));
```

Comment:
Ok, maybe last nits here: 
1. RE: hoisting, cssProcessorOptions should go above line 47
2. I think it's ok to do an equal on ALL of the options you are expecting, so I'd add back the `{ from, to }` business. (since that is something the plugin is doing)
3. line 51 should read `return Promise.resolve({ css: actualCss });` no?